samples/remote_attestation/common/dispatcher.cpp

```diff
@@ -234,16 +431,37 @@ int ecall_dispatcher::process_encrypted_msg(
     }
 
     data_size = sizeof(data);
-    if (m_crypto->decrypt(
-            encrypted_data, encrypted_data_size, data, &data_size))
+
+    memcpy(iv_str, &encrypted_data[ENCLAVE_SECRET_DATA_SIZE], IV_SIZE);
+    memcpy(
+        tag_str,
+        &encrypted_data[ENCLAVE_SECRET_DATA_SIZE + IV_SIZE],
+        ENCLAVE_SECRET_DATA_SIZE);
+
+    // Convert sequence number to 4 character bytes
+    m_enclave_config->sequence_number++;
+    add_str[0] = m_enclave_config->sequence_number & 0xff;
```

Comment:
Do you need this 4-line approach?  Maybe just use htonl?  E.g. something along the lines of
*((uint32_t*)add_str) = htonl(m_enclave_config->sequence_number);
I believe that allows the compiler the opportunity to optimize the byte swap to a single assembly instruction.

Reply:
Reason I added the 4-lines because it was clear which byte-order was being used. htonl is a good alternative which I hadn't considered. Thanks. Will incorporate.

Reply:
Reason I added the 4-lines because it was clear which byte-order was being used and the exact bytes that were being sent. Will consider your suggestion.

Reply:
I am not doing a byte swap but just converting the sequence number to 4 character bytes. Improved the code to just do a memcpy. This should work as both the enclaves are on the same machine. 
Adding htonl, etc. will complicate this sample and require including the socket libraries, etc.